near-sdk-sim/README.md

```diff
@@ -26,79 +32,197 @@ near-sdk = { git = "https://github.com/near/near-sdk-rs.git", tag="3.0.0" }
+
+**Want an example?** Check out [examples/fungible-token](../examples/fungible-token).
+
+
+## Test file
+
+In the root of your project (`contract-wrap` in the example above), create a `tests` directory with a Rust file inside. It's common to call this `tests/general.rs`.
+
+In this file you will need to include the bytes of the contract(s) you want to test:
 
 ```rust
-struct TokenContract {
-    account_id: String
-}
+near_sdk_sim::lazy_static_include::lazy_static_include_bytes! {
+    // update `contract.wasm` for your contract's name
+    CONTRACT_WASM_BYTES => "target/wasm32-unknown-unknown/release/contract.wasm",
 
-impl TokenContract {
-  pub fn transfer(&self, amount: U128) -> near_sdk::PendingContractTransaction {
-     //...
-  }
+    // if you run `cargo build` without `--release` flag:
+    CONTRACT_WASM_BYTES => "target/wasm32-unknown-unknown/debug/contract.wasm",
 }
 ```
 
-Thus you need to import this new type into your test file like so:
+Note that this means **you must `build` before you `test`!** Since `cargo test` does not re-generate the `wasm` files that your simulation tests rely on, you will need to `cargo build --all --target wasm32-unknown-unknown` before running `cargo test`. If you made contract changes and _you swear it should pass now_, try rebuilding!
+
+Now you can make a function to initialize your simulator:
 
 ```rust
-extern crate token;
-use token::TokenContract;
+use near_sdk_sim::{init_simulator, to_yocto, STORAGE_AMOUNT};
+
+fn init() {
+    // Use `None` for default genesis configuration
+    let root = init_simulator(None);
+
+    let contract = root.create_user(
+        CONTRACT_ID.to_string(), // account id
+        to_yocto("100"),         // initial balance
+    );
+
+    let contract = contract.deploy(
+        &CONTRACT_WASM_BYTES,
+        CONTRACT_ID.to_string(),
+        STORAGE_AMOUNT
+    );
+
+    (root, contract)
+}
 ```
 
-Next you need to import the simulator's init method and some other imports.
+Now you can start writing tests:
 
 ```rust
-use near_sdk_sim::{init_simulator, deploy, call, to_yocto};
+use near_sdk::serde_json::json;
+use near_sdk_sim::DEFAULT_GAS;
+
+#[test]
+fn simulate_some_view_function() {
+    let (root, contract) = init();
+
+    let actual: String = root.view(
+        CONTRACT_ID.to_string(),
+        "view_something".to_string(),
+        json!({
+            "some_param": "some_value".to_string(),
+        }).to_string().into_bytes()
+    ).unwrap_json();
+
+    assert_eq!("expected", actual);
+}
+
+#[test]
+fn simulate_some_change_method() {
+    let (root, contract) = init();
+
+    let result = root.call(
+        CONTRACT_ID.to_string(),
+        "change_something".to_string(),
+        json!({
+            "some_param": "some_value".to_string(),
+        }).to_string().into_bytes()
+        DEFAULT_GAS / 2,
+        0 // attached deposit
+    );
+
+    assert!(result.is_ok());
```

Comment:
@willemneal will this & the view call above work properly once https://github.com/near/near-sdk-rs/pull/325 is merged, or did I mess up some syntax?

Reply:
Correct, this should work with the PR mentioned.  We could also consider adding the README to the doc tests: https://doc.rust-lang.org/rustdoc/documentation-tests.html#include-items-only-when-collecting-doctests